core/src/main/java/org/apache/accumulo/core/classloader/ClassLoaderUtil.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.classloader;
+
+public class ClassLoaderUtil {
+
+  public static synchronized <U> Class<? extends U> loadClass(String contextName, String className,
```

Comment:
Now that you bring it up, I'm not sure that this needs to be synchronized.

Reply:
Looking into this overall issue a bit further in the existing code, iterators call AccumuloVFSClassLoader.loadClass() which is static and synchronized.  Also the iterator loading code has a class cache, maybe this was created to avoid the global synchronization.  Also I did not find an issue, will open one after looking into it a bit more.

Reply:
The cache is only narrowly used, so the synchronization is probably still problematic I'll open an issue.

Reply:
Created #1730